### PR TITLE
Handle case where a user has multiple roles on resources

### LINKED ISSUE #622: Handle case where a user has multiple roles on resources
This is a follow-up for https://github.com/fabric8-services/fabric8-auth/issues/545 in which an endpoint lists all resources of a given type for which the user has a role in. But the current response does not properly handles the case where the user has multiple roles on the same resource:

````
 {
     "data": [{
         "attributes": {
             "role": "contributor",
             "scopes": ["contribute", "view"]
         },
         "id": "18c25d66-b160-4c4f-9705-22f68dcca102",
         "type": "spaces"
     }, {
         "attributes": {
             "role": "admin",
             "scopes": ["contribute", "manage", "view"]
         },
         "id": "18c25d66-b160-4c4f-9705-22f68dcca102",
         "type": "spaces"
     }]
 }

````

### PR DESCRIPTION
**Pull request by xcoulon**
return an array of roles with its name and associated scopes
for each resource in which the user has a role in.

Eg:
````
{
    "data": [{
        "attributes": {
            "roles": [{
                "name": "contributor",
                "scopes": ["contribute", "view"]
            }, {
                "name": "admin",
                "scopes": ["contribute", "manage", "view"]
            }]
        },
        "id": "a41e48b7-369d-4316-a0a0-ba52c1e5ef55",
        "type": "spaces"
    }]
}
````

Fixes #622

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-627-1`

File: controller/user.go [hunk 1]
```diff
@@ -102,20 +102,41 @@ func (c *UserController) ListResources(ctx *app.ListResourcesUserContext) error
...
-		ID:   r.ResourceID,
-		Attributes: &app.UserResourceDataAttributes{
-			Role:   r.RoleName,
-			Scopes: r.Scopes,
-		},
+func convertToUserResourcesData(rolesPerResource map[string][]role.ResourceRoleDescriptor) []*app.UserResourceData {
+	result := make([]*app.UserResourceData, 0)
+	for resourceID, roles := range rolesPerResource {
+		data := &app.UserResourceData{
+			Type:       "spaces", // could be compared to r.ResourceType for a more generic response
```
Comment (sbose78) [inline controller/user.go]: I think we should call this something like type=resources?

**Comment by xcoulon** [comment]
Closing in favor of https://github.com/fabric8-services/fabric8-auth/issues/629 #RIP

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-627-2`

**Pull request closed without merging**